visualization-project-laptop-sales/visualization_project_laptop_sales_solution.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+"""visualization-project-laptop-sales_sol
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/10M53HHyac1sf2zjJ_UkB59movx5UKI0y
+"""
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import datetime
+
+"""# **#1 What is target market? Male or female?**"""
+
+sales = pd.read_csv(r'/content/Sales_data.txt')
+sales
+
+# Comparing 'Profit' between male and female
+total_profit = sales.groupby('Contact Sex')['Profit'].sum()
+total_profit
+
+"""1. The target marget seems to be males since their total profit of 2936.53 is higher than of females, which is 2522.48."""
+
+# Example, sale price of $500, your cost is $300, and the shipping is $10.
+# Case 1: You pass the shipping cost to the customer, so
+# profit = (500 + 10) - $300 = 210
+# Case 2: As the business, you take in the shipping cost
+# profit = 500 - (300 + 10) = 190
+
+"""# **#2 If business is cash-constrained, which gender should be targeted?**"""
+
+# Let's see what happens if the business takes on the shipping cost
+sales['Our Cost Plus Shipping'] = sales['Our Cost'] + sales['Shipping Cost']
+business_cost2 = sales.groupby('Contact Sex')['Our Cost Plus Shipping'].sum()
+business_cost2
+
+# Consider the profit percentage given the business takes on the shipping cost
+sales['Profit With Our Increased Cost'] = sales['Sale Price'] - sales['Our Cost Plus Shipping']
+sales['Our Profit Percentage With Our Increased Cost'] = (sales['Profit With Our Increased Cost'] / sales['Our Cost Plus Shipping']) * 100
+sales['Our Profit Percentage With Our Increased Cost']
+
+sales.groupby('Contact Sex')['Our Profit Percentage With Our Increased Cost'].mean()
+
+"""2. If the business is cash-constrained, they should target males.
+
+# **#3 If consumer is cash-constrained, which gender should be targeted?**
+"""
+
+# Let's see what happens if the the customer takes on the shipping cost
+sales['Sale Price Plus Shipping'] = sales['Sale Price'] + sales['Shipping Cost']
+consumer_cost2 = sales.groupby('Contact Sex')['Sale Price Plus Shipping'].sum()
+consumer_cost2
```

Comment:
You want to divide the Profit (grouped by contact sex) by this for the final answer. The below code is not needed.